src/firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.c

```diff
@@ -0,0 +1,68 @@
+#include "firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.h"
+
+#include <stdlib.h>
+
+typedef struct AllegroA3931MotorDriver
+{
+    PwmPin_t* pwm_pin;
+    GpioPin_t* reset_pin;
+    GpioPin_t* coast_pin;
+    GpioPin_t* mode_pin;
+    GpioPin_t* direction_pin;
+    GpioPin_t* brake_pin;
+    GpioPin_t* esf_pin;
+} AllegroA3931MotorDriver_t;
+
+AllegroA3931MotorDriver_t* io_allegro_a3931_motor_driver_create(
+    PwmPin_t* pwm_pin, GpioPin_t* reset_pin, GpioPin_t* coast_pin, GpioPin_t* mode_pin,
+    GpioPin_t* direction_pin, GpioPin_t* brake_pin, GpioPin_t* esf_pin)
+{
+    AllegroA3931MotorDriver_t* motor =
+        (AllegroA3931MotorDriver_t*)malloc(sizeof(AllegroA3931MotorDriver_t));
+
+    motor->pwm_pin       = pwm_pin;
+    motor->reset_pin     = reset_pin;
+    motor->coast_pin     = coast_pin;
+    motor->mode_pin      = mode_pin;
+    motor->direction_pin = direction_pin;
+    motor->brake_pin     = brake_pin;
+    motor->esf_pin       = esf_pin;
+
+    io_gpio_pin_setInactive(motor->reset_pin);
+    io_gpio_pin_setInactive(motor->coast_pin);
+    io_gpio_pin_setActive(motor->mode_pin);
+    // TODO: what direction is CW and CCW???
```

Comment:
TODO

Reply:
leaving this open, could you please tell simon to respond to my message? Need to have a quick call with him to do a final test run with this code and figure this out.

Reply:
This is resolved now. Effectively at startup there is no guarantee about the direction, the user should set it (verified that direction setting does work with Simon).

Reply:
(removed TODO)